c/parallel/src/transform.cu

```diff
@@ -692,10 +630,12 @@ CUresult cccl_device_transform_cleanup(cccl_device_transform_build_result_t* bui
     {
       return CUDA_ERROR_INVALID_VALUE;
     }
-    std::unique_ptr<char[]> cubin(reinterpret_cast<char*>(build_ptr->cubin));
-    std::unique_ptr<transform::runtime_tuning_policy> rtp(
-      reinterpret_cast<transform::runtime_tuning_policy*>(build_ptr->runtime_policy));
-    std::unique_ptr<transform::cache> cache(reinterpret_cast<transform::cache*>(build_ptr->cache));
+    using namespace cub::detail::transform;
+    std::unique_ptr<char[]> cubin(static_cast<char*>(build_ptr->cubin));
+    std::unique_ptr<arch_policies<1>> rtp(static_cast<arch_policies<1>*>(build_ptr->runtime_policy)); // FIXME(bgruber):
+                                                                                                      // handle <2> as
+                                                                                                      // well
```

Comment:
Is there any way in this function to distinguish whether we build the unary or binary transform?